modules/javafx.controls/src/main/java/javafx/scene/control/skin/MenuButtonSkinBase.java

```diff
@@ -214,6 +218,16 @@ public MenuButtonSkinBase(final C control) {
 
     /** {@inheritDoc} */
     @Override public void dispose() {
+        // FIXME : JDK-8244112 - backout if we are already disposed
+        // should check for getSkinnable to be null and return
+
+        // Cleanup accelerators
+        if (getSkinnable().getScene() != null) {
+            ControlAcceleratorSupport.removeAcceleratorsFromScene(getSkinnable().getItems(), getSkinnable().getScene());
+        }
+
+        // Remove listeners
+        getSkinnable().sceneProperty().removeListener(sceneChangeListener);
```

Comment:
If scene is set to null (which can happen if this control or its parent is removed from the scene graph) before dispose is called, this will throw an NPE. I think it should be moved inside the above if statement.

This raises the question: when the scene changes, are both the accelerators and this listener removed from the old scene?

Reply:
Yes, you are of course right about `sceneProperty()` not being null just because scene might be. An my other question was based on a too-quick read of the code: since the listener is on the scene property of the control, and is meant to track the case where the control moves from one scene to another, it would not be the right thing to do to remove it when changing scenes.

So the fix looks fine to me.